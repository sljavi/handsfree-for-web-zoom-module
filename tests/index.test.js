import module, {
  zoomInTab,
  zoomOutTab,
  restoreZoom
} from '../src/index'

const defaultTabs = [{
  id: 1
}]

const zoom = 1.2

function mockExtension(methods = {}) {
  global.chrome = {
    extension: {
      sendMessage: jest.fn(),
      ...methods
    },
    tabs: {
      query: (query, cb) => cb(defaultTabs),
      getZoom: (tabId, cb) => cb(zoom),
      setZoom: jest.fn(),
      ...methods
    }
  }
}

describe('Module', () => {
  it('defines a module', () => {
    expect(module).toMatchSnapshot()
  })
  describe('zoomOut action', () => {
    it('calls background action', () => {
      const sendMessage = jest.fn()
      mockExtension({ sendMessage })
      const zoomOut = module.contexts[0].commands[0]
      expect(zoomOut.name).toEqual('i18n-command.zoom-out')
      zoomOut.action()
      expect(sendMessage.mock.calls).toMatchSnapshot()
    })
  })
  describe('zoomIn action', () => {
    it('calls background action', () => {
      const sendMessage = jest.fn()
      mockExtension({ sendMessage })
      const zoomIn = module.contexts[0].commands[1]
      expect(zoomIn.name).toEqual('i18n-command.zoom-in')
      zoomIn.action()
      expect(sendMessage.mock.calls).toMatchSnapshot()
    })
  })
  describe('resetZoom action', () => {
    it('calls background action', () => {
      const sendMessage = jest.fn()
      mockExtension({ sendMessage })
      const resetZoom = module.contexts[0].commands[2]
      expect(resetZoom.name).toEqual('i18n-command.reset-zoom')
      resetZoom.action()
      expect(sendMessage.mock.calls).toMatchSnapshot()
    })
  })
})

describe('zoomInTab background action', () => {
  it('increases the zoom level', () => {
    const setZoom = jest.fn()
    mockExtension({ setZoom })
    zoomInTab()
    expect(setZoom).toHaveBeenCalledWith(1, zoom + 0.1)
  })
})

describe('zoomOutTab background action', () => {
  it('decreases the zoom level', () => {
    const setZoom = jest.fn()
    mockExtension({ setZoom })
    zoomOutTab()
    expect(setZoom).toHaveBeenCalledWith(1, zoom - 0.1)
  })
})

describe('restoreZoom background action', () => {
  it('restore the zoom level', () => {
    const setZoom = jest.fn()
    mockExtension({ setZoom })
    restoreZoom()
    expect(setZoom).toHaveBeenCalledWith(1, 0)
  })
})
