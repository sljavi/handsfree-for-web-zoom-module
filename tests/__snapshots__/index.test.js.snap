// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module defines a module 1`] = `
Object {
  "contexts": Array [
    Object {
      "commands": Array [
        Object {
          "action": [Function],
          "group": "i18n-group.zoom",
          "help": "i18n-help.zoom-out",
          "name": "i18n-command.zoom-out",
        },
        Object {
          "action": [Function],
          "group": "i18n-group.zoom",
          "help": "i18n-help.zoom-in",
          "name": "i18n-command.zoom-in",
        },
        Object {
          "action": [Function],
          "group": "i18n-group.zoom",
          "help": "i18n-help.reset-zoom",
          "name": "i18n-command.reset-zoom",
        },
      ],
      "context": "root",
      "i18n": Object {
        "en": Object {
          "command.reset-zoom": "reset zoom",
          "command.zoom-in": "zoom in",
          "command.zoom-out": "zoom out",
          "group.zoom": "Zoom",
          "help.reset-zoom": "Restore the zoom to the default level",
          "help.zoom-in": "Zooms in the current tab",
          "help.zoom-out": "Zooms out the current tab",
        },
        "es": Object {
          "command.reset-zoom": "restaurar",
          "command.zoom-in": "agrandar",
          "command.zoom-out": "achicar",
          "group.zoom": "Zoom",
          "help.reset-zoom": "Restaura el tamaño normal del sitio",
          "help.zoom-in": "Incrementa el tamaño del contenido del sitio",
          "help.zoom-out": "Decrementa el tamaño del contenido del sitio",
        },
        "pt": Object {
          "command.reset-zoom": "resetar zoom",
          "command.zoom-in": "aumentar zoom",
          "command.zoom-out": "diminuir zoom",
          "group.zoom": "Zoom",
          "help.reset-zoom": "Restaura o zoom para o estado original da guia",
          "help.zoom-in": "Aumenta o zoom na guia atual",
          "help.zoom-out": "Diminuir o zoom na guia atual",
        },
      },
    },
  ],
  "description": "i18n-description",
  "i18n": Object {
    "en": Object {
      "description": "Commands useful to zoom in and zoom out the website.",
      "name": "Zoom",
    },
    "es": Object {
      "description": "Comandos útiles para cambiar el tamaño de los contentidos del sitio web.",
      "name": "Zoom",
    },
    "pt": Object {
      "description": "Comando utilizado para aumentar e diminuir o zoom na guia.",
      "name": "Zoom",
    },
  },
  "icon": "fa fa-search-plus",
  "name": "i18n-name",
}
`;

exports[`Module resetZoom action calls background action 1`] = `
Array [
  Array [
    Object {
      "resetZoom": true,
    },
    [Function],
  ],
]
`;

exports[`Module zoomIn action calls background action 1`] = `
Array [
  Array [
    Object {
      "zoomIn": true,
    },
    [Function],
  ],
]
`;

exports[`Module zoomOut action calls background action 1`] = `
Array [
  Array [
    Object {
      "zoomOut": true,
    },
    [Function],
  ],
]
`;
